test/unit/test_cache.py

```diff
@@ -461,81 +474,80 @@ def test_load_with_expired_entries(self, tmpdir):
         # cache will expire immediately due to the NO_LIFETIME setting
         # when loading again, clearing cache logic will be triggered
         # load will trigger clear expired entries, and then further call _getitem
-        c2._load()
+        c2.load()
 
         assert len(c2) == 1 and c2["a"] == 1 and c2._getitem_non_locking("a") == 1
 
     def test_broken_pickle_objects(self, tmpdir):
         cache_path = os.path.join(tmpdir, "cache.txt")
-        c1 = cache.SFDictFileCache(file_path=cache_path)
+        c1 = NeverSaveSFDictFileCache(file_path=cache_path)
         c1["key"] = BrokenReadingPickleObject()
-        # __setitem__ might have already saved the item, so cache is unchanged after saving, thus we force flush here
-        assert c1._save(force_flush=True)
+        assert c1.save()
         assert os.path.exists(cache_path) and os.path.isfile(cache_path)
 
-        c2 = cache.SFDictFileCache(file_path=cache_path)
+        c2 = NeverSaveSFDictFileCache(file_path=cache_path)
         assert not c2._load()  # load should return false due to ModuleNotFound error
 
         cache_path = os.path.join(tmpdir, "cache2.txt")
-        c1 = cache.SFDictFileCache(file_path=cache_path)
+        c1 = NeverSaveSFDictFileCache(file_path=cache_path)
         c1["key"] = BrokenWritingPickleObject()
-        assert not c1._save()
+        assert not c1.save()
         assert not os.path.exists(cache_path)
 
-        c2 = cache.SFDictFileCache(file_path=cache_path)
-        assert not c2._load()  # load should return false due to no file
+        c2 = NeverSaveSFDictFileCache(file_path=cache_path)
+        assert not c2.load()  # load should return false due to no file
 
 
 def test_file_is_not_updated(tmpdir):
     tmp_cache_file = os.path.join(tmpdir, "tmp_cache")
-    sfcache = cache.SFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
+    sfcache = NeverSaveSFDictFileCache(file_path=tmp_cache_file, entry_lifetime=1)
```

Comment:
I think we can revert the change here, the purpose of this test is to test that when calling `update`, sfcache doesn't try to save the cache.